### PR TITLE
Fix RAPS Endpoints 

### PR DESCRIPTION
**Pull request by ebwinters**
RAPS added redirects which put a unique cookie in the header. To get around this I'm using the tough cookie and cookie jar libraries.

### CODE REVIEW

**Comment by EliteDaMyth** [comment]
Very cool bypass! Looks fine to me.

**Review by EliteDaMyth — DISMISSED** [review]

**Comment by ebwinters** [comment]
Tests were passing locally SMH

File: scrapers/covid-19/getVaccine.js [hunk 1]
```diff
@@ -26,24 +30,49 @@ const cleanData = (data) => {
 const getVaccineData = async (keys, redis) => {
 	let day, month, year;
 	try {
-		const html = cheerio.load((await axios.get('https://www.raps.org/news-and-articles/news-articles/2020/3/covid-19-vaccine-tracker')).data);
+		const cookieJar = new tough.CookieJar();
```
Comment (puf17640) [inline scrapers/covid-19/getVaccine.js:33]: do we need to create a new cookie jar every time we run getVaccineData?

File: scrapers/covid-19/getVaccine.js [hunk 2]
```diff
@@ -26,24 +30,49 @@ const cleanData = (data) => {
...
-			data: cleanData(parsedData)
-		}));
-	} catch (err) {
-		logger.err('Error: Requesting vaccine CSV data failed!', err);
-	}
+	let dataExists = false;
+	let counter = 0;
+	do {
+		try {
+			const cookieJar = new tough.CookieJar();
```
Comment (puf17640) [inline scrapers/covid-19/getVaccine.js:52]: same question as above

File: scrapers/covid-19/getVaccine.js [hunk 3]
```diff
@@ -26,24 +30,49 @@ const cleanData = (data) => {
...
+			console.log(`https://www.raps.org/RAPS/media/news-images/data/${year}${months[month]}${(day - counter).toString().padStart(2, '0')}-vax-tracker-Craven.csv`);
```
Comment (puf17640) [inline scrapers/covid-19/getVaccine.js:53]: might wanna remove the console.log

File: scrapers/covid-19/getVaccine.js [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
+const axiosCookieJarSupport = require('axios-cookiejar-support').default;
+const tough = require('tough-cookie');
 const csv = require('csvtojson');
 const cheerio = require('cheerio');
 const logger = require('../../utils/logger');
 const { months, phaseData } = require('../../utils/RAPS');
 
+axiosCookieJarSupport(axios);
+
 const cleanData = (data) => {
```
Comment (puf17640) [inline scrapers/covid-19/getVaccine.js:11]: why dont we add this to /utils/RAPS

**Review by puf17640 — DISMISSED** [review]
im fine with it, check comments for feedback 👍

File: scrapers/covid-19/getVaccine.js (hunk 4 above)
Comment (ebwinters) [reply scrapers/covid-19/getVaccine.js:11]: If the fix works I'll do it at some point later in the week, a lot of code reuse between vaccine/therepeutics you are right

Comment (ebwinters) [reply scrapers/covid-19/getVaccine.js:10]: made this an issue #949 doing it later time

**Pull request merged**